src/engine/guidance/lane_processing.cpp

```diff
@@ -74,34 +75,110 @@ std::vector<RouteStep> anticipateLaneChange(std::vector<RouteStep> steps,
             if (!lanes_to_constrain || !lanes_fan_in)
                 return;
 
-            // In case there is no lane information we work with one artificial lane
-            const auto current_adjusted_lanes = std::max(current_lanes.lanes_in_turn, LaneID{1});
+            // We do not have a mapping from lanes to lanes. All we have is the lanes in the turn
+            // and all the lanes at that situation. Make the best out of it.
```

Comment:
could you add some explanation what would be required to improve this and how an ideal solution would look like?